src/Service.Tests/Unittests/MultipleCreateUnitTests/MultipleCreateOrderHelperUnitTests.cs

```diff
@@ -0,0 +1,399 @@
+
+            // Test 2: The value for relationship field 'username' is present in the input for the target entity.
+            // Setup column input in source entity.
+            //
+            // The complete graphQL mutation looks as follows:
+            //  mutation{
+            //          createUser_NonAutogenRelationshipColumn(item: {
+            //            email: "dab@microsoft.com",
+            //            UserProfile_NonAutogenRelationshipColumn: {
+            //              profilepictureurl: "dab/profilepicture",
+            //              userid: 10,
+            //              username: "DAB"
+            //            }
+            //          }){
+            //            <selection_set>   
+            //          }
+            //      }
+            columnDataInSourceBody = new()
+            {
+                { "email", new StringValueNode("dab@microsoft.com") }
+            };
+
+            // Setup column input in target entity.
+            targetNodeValue = new();
+            fields = new()
+            {
+                new ObjectFieldNode("profilepictureurl", "dab/profilepicture"),
+                new ObjectFieldNode("userid", 10),
+                new ObjectFieldNode("username", "DAB")
+            };
+            targetNodeValue = targetNodeValue.WithFields(fields);
+
+            // Get the referencing entity name.
+            referencingEntityName = MultipleCreateOrderHelper.GetReferencingEntityName(
+                relationshipName: "UserProfile_NonAutogenRelationshipColumn",
+                context: context,
+                sourceEntityName: sourceEntityName,
+                targetEntityName: targetEntityName,
+                metadataProvider: _sqlMetadataProvider,
+                columnDataInSourceBody: columnDataInSourceBody,
+                targetNodeValue: targetNodeValue,
+                nestingLevel: 1);
+            // Since the target entity contained the value for relationship field,
+            // it act as the referenced entity, and the source entity act as the referencing entity.
+            Assert.AreEqual(sourceEntityName, referencingEntityName);
+        }
+
+        /// <summary>
+        /// Test to validate that when a pair of relationship fields in the source/target entity are autogenerated,
+        /// it is not possible to determine a referencing entity. This is because we cannot provide a value for insertion
+        /// for an autogenerated relationship field in any of the entity.
+        /// Hence, none of the entity can act as a referencing/referenced entity.
+        ///
+        /// Here, the relationship between entities: 'User_AutogenRelationshipColumn' and 'UserProfile_AutogenRelationshipColumn'
+        /// is defined as User_AutogenRelationshipColumn(userid) -> UserProfile_AutogenRelationshipColumn(profileid)
+        /// where both the relationships fields 'username' are autogenerated in the respective entities.
```

Comment:
> where both the relationships fields 'username' are autogenerated in the respective entities.
What does `userName` have to do with this example? isn't the failure because the relationship is defined between two autogenerated columns, thus telling dab that both `User_...` and `UserProfile_...` entities want to be created first.. which is indeterministic?

Reply:
Right, username has nothing to do. Incorrect comment. Updated.

Reply:
Removed username field as well from the input since it isn't required.